src/Renci.SshNet/SshClient.cs

```diff
@@ -353,6 +371,37 @@ public Shell CreateShell(Encoding encoding, string input, Stream output, Stream
             return CreateShell(_inputStream, output, extendedOutput, terminalName, columns, rows, width, height, terminalModes, bufferSize);
         }
 
+        /// <summary>
+        /// Creates the shell.
+        /// </summary>
+        /// <param name="encoding">The encoding to use to send the input.</param>
+        /// <param name="input">The input.</param>
+        /// <param name="output">The output.</param>
+        /// <param name="extendedOutput">The extended output.</param>
+        /// <param name="bufferSize">Size of the internal read buffer.</param>
+        /// <returns>
+        /// Returns a representation of a <see cref="Shell" /> object.
+        /// </returns>
+        /// <exception cref="SshConnectionException">Client is not connected.</exception>
+        public Shell CreateShell(Encoding encoding, string input, Stream output, Stream extendedOutput, int bufferSize)
```

Comment:
Or rename it to `CreateShellNoTerminal`?

Reply:
Or `CreateShellWithoutTerminal`?

Reply:
Yes, it might be safest if the new functionality is exposed via `CreateShellNoTerminal` and `CreateShellStreamNoTerminal` just to make a clear distinction.

The overload on this particular line does not seem very useful to me, but if you use it you can keep it.